### ISSUE TITLE
AttributeError: 'DataFrame' object has no attribute 'dtype' when running setup

### ISSUE THREAD

**Issue by KuriaMaingi** [pycaret/pycaret#240]
Receiving error below executing this line of code:
`from pycaret import regression
exp_reg1 = regression.setup(data = dfcat, target = 'val_chg1_perc', session_id=1)`

Running python=3.6.10, pandas=1.0.5, pycaret=1.0.0

> Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "~\anaconda3\envs\dspy36pycaret\lib\site-packages\pycaret\regression.py", line 957, in setup
    random_state = seed)
  File "~\envs\dspy36pycaret\lib\site-packages\pycaret\preprocess.py", line 2540, in Preprocess_Path_One
    return(pipe.fit_transform(train_data))
  File "~\envs\dspy36pycaret\lib\site-packages\sklearn\pipeline.py", line 383, in fit_transform
    Xt, fit_params = self._fit(X, y, **fit_params)
  File "~\envs\dspy36pycaret\lib\site-packages\sklearn\pipeline.py", line 313, in _fit
    **fit_params_steps[name])
  File "~\envs\dspy36pycaret\lib\site-packages\joblib\memory.py", line 352, in __call__
    return self.func(*args, **kwargs)
  File "~\envs\dspy36pycaret\lib\site-packages\sklearn\pipeline.py", line 726, in _fit_transform_one
    res = transformer.fit_transform(X, y, **fit_params)
  File "~\envs\dspy36pycaret\lib\site-packages\pycaret\preprocess.py", line 330, in fit_transform
    data = self.fit(data)
  File "~\envs\dspy36pycaret\lib\site-packages\pycaret\preprocess.py", line 111, in fit
    if data[i].dtype=='bool':
  File "~\envs\dspy36pycaret\lib\site-packages\pandas\core\generic.py", line 5274, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'dtype'

**Comment by pycaret** [comment]
@KuriaMaingi Thanks for reporting. Can you update your pandas to most recent version. 

Let me know if that works!

**Comment by KuriaMaingi** [comment]
Interestingly, I attached a debugger as well as added a try...except to catch the AttributeError around preprocess.py line 111 to understand which columns were causing this issue.

The issue was being caused by duplicate columns (same col name). I deduplicated the dataset and everything works swimmingly. It might be worth to add a callout in the documentation warning against this OR catching the exception.

This was with pandas 1.0.5.

Hope that is helpful!
